### PR TITLE
[swappable.requirements] Requires should now be Preconditions

### PR DESCRIPTION
**Pull request by BRevzin**

### CODE REVIEW

File: source/lib-intro.tex [hunk 1]
```diff
@@ -1614,15 +1614,15 @@
 \begin{codeblock}
 #include <utility>
 
-// Requires: \tcode{std::forward<T>(t)} shall be swappable with \tcode{std::forward<U>(u)}.
+// Mandates: \tcode{std::forward<T>(t)} shall be swappable with \tcode{std::forward<U>(u)}.
```
Comment (tkoeppe) [inline source/lib-intro.tex:1617]: Shouldn't we then also get rid of "shall", since the "mandates" style implies that?

**Review by tkoeppe — CHANGES_REQUESTED** [review]

File: source/lib-intro.tex (hunk 1 above)
Comment (jwakely) [reply source/lib-intro.tex:1617]: Yes, a Mandates condition is just stated. "is swappable with".

**Review by tkoeppe — APPROVED** [review]

**Review by jensmaurer — APPROVED** [review]

**Comment by jensmaurer** [comment]
When merging, fix the commit message to mention the section label.

File: source/lib-intro.tex [hunk 2]
```diff
@@ -1614,15 +1614,15 @@
 \begin{codeblock}
 #include <utility>
 
-// Requires: \tcode{std::forward<T>(t)} shall be swappable with \tcode{std::forward<U>(u)}.
+// Mandates: \tcode{std::forward<T>(t)} is swappable with \tcode{std::forward<U>(u)}.
```
Comment (CaseyCarter) [inline source/lib-intro.tex:1617]: Both "is swappable with" and "swappable" have semantic requirements, violations of which an implementation cannot be expected to diagnose. To be pedantically correct, these comments both need to Mandate the syntactic requirements and Precondition the semantic requirements. (This admittedly feels like overkill for an example - I have no better idea, though.)

Comment (tkoeppe) [reply source/lib-intro.tex:1617]: How about "requires:"? :-)

Comment (CaseyCarter) [reply source/lib-intro.tex:1617]: TFW you remove a term for being too vague and then complain about having to be too precise.

Comment (tkoeppe) [reply source/lib-intro.tex:1617]: Given that getting all those requirements right is critical and that this example concerns a presumably very common customization point, it might be worth doing this right.

@BRevzin: would you mind adding appropriate "precondition" comments?

**Review by JohelEGP — CHANGES_REQUESTED** [review]
Please, prefix the [stable.label] to the commit and issue title.

File: source/lib-intro.tex (hunk 2 above)
Comment (timsong-cpp) [reply source/lib-intro.tex:1617]: Or we can just use _Preconditions:_, which is how we do this elsewhere in the library spec.

File: source/lib-intro.tex [hunk 3]
```diff
@@ -1614,15 +1614,15 @@
 \begin{codeblock}
 #include <utility>
```
Comment (timsong-cpp) [inline source/lib-intro.tex:1615]: Drive-by: since this is an unusually complete example trying to illustrate what user code would look like, perhaps we should `#include <cassert>` here?

**Comment by jensmaurer** [comment]
@tkoeppe, this looks good now.

**Comment by tkoeppe** [comment]
Yes, I thought I'd let @timsong-cpp take a final look.

**Review by timsong-cpp — APPROVED** [review]
